### ISSUE TITLE
[Bug]: ReplainGain on some M4A files not detected

### ISSUE THREAD

**Issue by marksdestiny** [navidrome/navidrome#2679]
### I confirm that:

- [X] I have searched the existing [open AND closed issues](https://github.com/navidrome/navidrome/issues?q=is%3Aissue) to see if an issue already exists for the bug I've encountered
- [X] I'm using the latest version (your issue may have been fixed already)

### Version

0.50.1

### Current Behavior

The replay gain of some M4A files is not detected. 

This seems to the case when the name of the corresponding tag (REPLAYGAIN_TRACK_GAIN) is in all uppercase instead of all lowercase. As an example, the application [rsgain](https://github.com/complexlogic/rsgain) creates the tags with names that are all uppercase, and so does the application MusicBrainz Picard.

### Expected Behavior

The replay gain is correctly detected, as stored in the metadata of the M4A file.

### Steps To Reproduce

1. Run Navidrome 0.50.1 with its default settings.
2. Add the test audio files from [testdata.zip](https://github.com/navidrome/navidrome/files/13537841/testdata.zip) to the music library.
3. Login to the Navidrome UI and perform a full scan.
4. Check the details of the two tracks using the _Get Info_ context menu. The track _Working_ has the correct track replay gain (-6.1 dB). the track _Not Working_ has an incorrect track replay gain (0 dB).

### Environment

```markdown
- OS: Ubuntu 22.04
- Browser: Firefox 120.0
- Client: Navidrome UI
```

### How Navidrome is installed?

Docker

### Relevant log output

```shell
time="2023-12-03T02:04:06Z" level=trace msg="TagLib: read tags" filename=/music/notworking.m4a id=0 tags="map[bitrate:[192] channels:[2] date:[2023-12-03 2023] duration:[1] encodedby:[Lavf58.76.100] lengthinmilliseconds:[1000] title:[Not Working Not Working]]"
time="2023-12-03T02:04:06Z" level=trace msg="TagLib: read tags" filename=/music/working.m4a id=1 tags="map[bitrate:[192] channels:[2] date:[2023-12-03 2023] duration:[1] encodedby:[Lavf58.76.100] lengthinmilliseconds:[1000] replaygain_track_gain:[-6.10 dB] title:[Working Working]]"
```

### Anything else?

The solutions seems trivial, I might create a pull request.

### Code of Conduct

- [X] I agree to follow Navidrome's Code of Conduct

**Comment by deluan** [comment]
> This seems to the case when the name of the corresponding tag (REPLAYGAIN_TRACK_GAIN) is in all uppercase instead of all lowercase.

Yeah, that seems to be what is happening. RG_TAGS is a list of lower cased tags: https://github.com/navidrome/navidrome/blob/3debd31b1281672bd31f3becc5efc6615c6d2ecd/scanner/metadata/taglib/taglib_wrapper.cpp#L19-L23

We need to change the way we match the RG_TAGS here:
https://github.com/navidrome/navidrome/blob/3debd31b1281672bd31f3becc5efc6615c6d2ecd/scanner/metadata/taglib/taglib_wrapper.cpp#L87-L99

A PR is mostly welcome, but the change is in the CPP TagLib wrapper, not in the Go code.

c/c @kgarner7
